[ENG-3865][docs] add docs on running detox tests on EAS Build (#18696)
expo/expo: docs/pages/build-reference/tests.md

---
title: Running E2E tests on EAS Build
sidebar_title: Running E2E tests
---

import { Collapsible } from '~/ui/components/Collapsible';
import { Terminal } from '~/ui/components/Snippet';

> 

Reviewer: ```suggestion

> **Warning**: EAS Build support for E2E testing is in a *very early* state. The intention of this guide is to explain how you can run E2E tests on the service today, without all of the affordances that we plan to build in the future. This guide will evolve over time as support for testing workflows in EAS Build improves.

 >  **Android not yet supported**: EAS Build does not yet support running tests on an Android Emulator — this will be coming soon.
 
```

With EAS Build, you can build a workflow for running E2E tests for your application. In this guide, you will learn how to use one of the most popular libraries ([Detox](https://wix.github.io/Detox)) to do that.

> > This guide assumes that you have a **Bare Workflow** project.

Reviewer: ```suggestion
This guide explains how to run E2E tests with Detox in a bare workflow project. You can use [@config-plugins/detox](https://github.com/expo/config-plugins/tree/main/packages/detox) for a managed project, but you may need to adjust some of the instructions in this guide in order to do so.
```

> > **Warning**: Currently, EAS Build does not support running tests on Android Emulator. More information on this is comming soon.
> 

Reviewer: ```suggestion
```

## Running iOS tests

### 0. Initialize a new Bare Workflow project

Let's start by initializing a fresh Expo project and running `expo prebuild` to convert it to a Bare Workflow project.

<Terminal cmd={[
'# Initialize a new project',
'$ yarn create expo-app eas-tests-example',
'# cd into the project directory',
'$ cd eas-tests-example',
'# Generate native code',
'$ yarn expo prebuild --platform ios'
]} />

### 1. Make home screen interactive

> The first step to writing E2E tests is to make our app interactive. Let's add a button and handle the press event to display some text.

Reviewer: ```suggestion
The first step to writing E2E tests is to have something to test — we have an empty app, so let's make our app interactive. We can add a button and display some new text when it's pressed.
```
